modules/nf-core/sratools/prefetch/templates/retry_with_backoff.sh

```diff
@@ -47,7 +47,16 @@ retry_with_backoff !{args2} \
     !{args} \
     !{id}
 
-[ -f !{id}.sralite ] && vdb-validate !{id}.sralite || vdb-validate !{id}
+# check file integrity using md5sum or vdb-validate as fallback
+MD5SUM=$(
+	curl -s 'https://locate.ncbi.nlm.nih.gov/sdl/2/retrieve?filetype=run&acc=!{id}' |
+	grep -o '"md5": "[^"]*"' | cut -f4 -d'"'
```

Comment:
Since the response is JSON, it might be neater to replace the parsing with `jq`, like so.

```suggestion
	jq '.result[0] | .files[0] | .md5'
```

Of course, then we also need that as a dependency.

Reply:
I have thought about this. I didn't want to use `jq`, because it's not part of the container. I agree it would be the cleaner approach and could add it. However: I just learned that the response can contain multiple MD5 sums (in cases where multiple file formats are available for the given SRA ID, for example BAM **and** FastQ). In light of this, what do you think about taking the whole of the HTTP response as a plain string and simply `grep`ing it for the MD5 sum we calculate on the prefetched file? This way, we don't need `jq` and we will for sure find a match if there is one. And we avoid rather complicated code to take apart the nested JSON object and looping over all MD5 sums.